Assets/MixedRealityToolkit/AudioManager/AudioManager.cs

```diff
@@ -0,0 +1,310 @@
+        /// </summary>
+        /// <param name="emitterObject">The GameObject the AudioSource needs to be attached to</param>
+        /// <returns>An AudioSource reference if one exists, otherwise null</returns>
+        private static AudioSource GetUnusedSource(GameObject emitterObject)
+        {
+            ClearNullAudioSources();
+
+            for (int i = 0; i < AvailableSources.Count; i++)
+            {
+                AudioSource tempSource = AvailableSources[i];
+                if (tempSource.gameObject == emitterObject)
+                {
+                    return tempSource;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Remove any references to AudioSource components that no longer exist
+        /// </summary>
+        private static void ClearNullAudioSources()
+        {
+            for (int i = AvailableSources.Count - 1; i >= 0; i--)
+            {
+                AudioSource tempSource = AvailableSources[i];
+                if (tempSource == null)
+                {
+                    AvailableSources.RemoveAt(i);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Make sure that the AudioManager has all of the required components
+        /// </summary>
+        /// <returns>Whether there is a valid AudioManager instance</returns>
+        private static bool ValidateManager()
+        {
+            if (Instance == null)
+            {
+                CreateInstance();
+                if (Instance == null)
+                {
+                    return false;
+                }
+                else
+                {
+                    return true;
+                }
+            }
+
+            return true;
+        }
+
+        #endregion
+
+        #region Editor
+
+        /// <summary>
+        /// Mute all ActiveEvents that are not soloed
+        /// </summary>
+        public static void ApplyActiveSolos()
+        {
+            ValidateManager();
+
+            bool soloActive = false;
+            for (int i = 0; i < ActiveEvents.Count; i++)
+            {
+                if (ActiveEvents[i].Soloed)
+                {
+                    soloActive = true;
+                }
+            }
+
+            if (soloActive)
+            {
+                for (int i = 0; i < ActiveEvents.Count; i++)
+                {
+                    ActiveEvents[i].ApplySolo();
+                }
+            }
+            else
+            {
+                ClearActiveSolos();
+            }
+        }
+
+        /// <summary>
+        /// Unmute all events
+        /// </summary>
+        public static void ClearActiveSolos()
+        {
+            ValidateManager();
+
+            for (int i = 0; i < ActiveEvents.Count; i++)
+            {
+                ActiveEvents[i].ClearSolo();
+            }
+        }
+
+        #endregion
+    }
+
+    public enum AudioLanguage
```

Comment:
please move enums into their own files.

Reply:
This enum could likely be more generic than just for audio?

Reply:
perhaps use the .net culture codes instead (ex: en-US, fr-CA, etc). that would allow for easier globalization (we will need this type of support for text soon as well).

Reply:
I agree it would be more useful as a generic selection. Would that change where it lives?

Reply:
Not sure a separate enum for AudioLanguage would be needed with this suggestion. Instead, code that currently references AudioLanguage would use the .net CultureInfo class.

https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo?view=netframework-4.7.2

While it's a bit more work for the developer to create a CultureInfo, it enables the toolkit to leverage the .net globalization features more seamlessly.